cloud/aws/aws_env.cc

```diff
@@ -221,38 +222,86 @@ Aws::S3::Model::HeadObjectOutcome AwsS3ClientWrapper::HeadObject(
   return outcome;
 }
 
+static bool GetNameFromEnvironment(const char *name, const char *alt, std::string * result) {
+  char *value = getenv(name);
+  if (value == nullptr && alt != nullptr) {
+    value = getenv(alt);
+  }
+  if (value != nullptr) {
+    result->assign(value);
+    return true;
+  } else {
+    return false;
+  }
+}
+  
+static std::string GetFullBucketName(const std::string & name, const char *bucket) {
+  std::string full_name;
+  if (! GetNameFromEnvironment(bucket, "ROCKSDB_CLOUD_BUCKET_NAME", &full_name)) {
+    full_name = name;
+  }
+  if (!full_name.empty()) {
+    char* pname = getenv("ROCKSDB_CLOUD_BUCKET_PREFIX");
+    if (pname != nullptr) {
+      full_name = trim(pname) + full_name;
+    } else {
+      full_name = "rockset." + full_name;
+    }
+  }
+  return full_name;
+}
 //
 // The AWS credentials are specified to the constructor via
 // access_key_id and secret_key.
 //
-AwsEnv::AwsEnv(Env* underlying_env, const std::string& src_bucket_prefix,
-               const std::string& src_object_prefix,
+AwsEnv::AwsEnv(Env* underlying_env,
+	       const std::string& src_bucket_name,
+               const std::string& src_object_path,
                const std::string& src_bucket_region,
-               const std::string& dest_bucket_prefix,
-               const std::string& dest_object_prefix,
+               const std::string& dest_bucket_name,
+               const std::string& dest_object_path,
                const std::string& dest_bucket_region,
                const CloudEnvOptions& _cloud_env_options,
                std::shared_ptr<Logger> info_log)
     : CloudEnvImpl(_cloud_env_options.cloud_type, _cloud_env_options.log_type,
                    underlying_env),
       info_log_(info_log),
       cloud_env_options(_cloud_env_options),
-      src_bucket_prefix_(src_bucket_prefix),
-      src_object_prefix_(src_object_prefix),
-      src_bucket_region_(src_bucket_region),
-      dest_bucket_prefix_(dest_bucket_prefix),
-      dest_object_prefix_(dest_object_prefix),
-      dest_bucket_region_(dest_bucket_region),
       running_(true),
       has_src_bucket_(false),
       has_dest_bucket_(false),
       dest_equal_src_(false) {
-  src_bucket_prefix_ = trim(src_bucket_prefix_);
-  src_object_prefix_ = trim(src_object_prefix_);
-  src_bucket_region_ = trim(src_bucket_region_);
-  dest_bucket_prefix_ = trim(dest_bucket_prefix_);
-  dest_object_prefix_ = trim(dest_object_prefix_);
-  dest_bucket_region_ = trim(dest_bucket_region_);
+  Aws::InitAPI(Aws::SDKOptions());
+  src_bucket_name_   = GetFullBucketName(src_bucket_name, "ROCKSDB_CLOUD_SRC_BUCKET_NAME");
+  src_object_path_ = trim(src_object_path);
+  dest_bucket_name_  = GetFullBucketName(dest_bucket_name, "ROCKSDB_CLOUD_DEST_BUCKET_NAME");
+  dest_object_path_ = trim(dest_object_path);
```

Comment:
Feel free to remove the trim business here.

Reply:
The values were trimmed before.  I put the trim in here because of that.  Am removing per your suggestion.